packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts

```diff
@@ -70,7 +70,10 @@ export function serverBareModulesPlugin(
         if (
           onWarning &&
           !isNodeBuiltIn(packageName) &&
-          !/\bnode_modules\b/.test(importer)
+          !/\bnode_modules\b/.test(importer) &&
+          // silence spurious warnings when using Yarn PnP until issues with
+          // ambiguous/unsound extraneous dependencies are fixed
+          process.versions.pnp == null
```

Comment:
If I understand this correctly, we're currently looking exclusively for Node built-ins or in `node_modules/` for dependencies, but Yarn PnP doesn't put things into `node_modules/`, so our existing check fails and thinks the dependency is missing.

@lensbart is that an accurate paraphrasing?

If so, could you include something to that effect in the comment here explaining _why_ checking `node_modules/` is insufficient for Yarn PnP?

Reply:
Also instead of _disabling_ warnings for Yarn PnP, it'd be better to _enable_ them for compatible package managers.

Reply:
@pcattori Updated the comment!

Reply:
Not sure how to check for other package managers such as pnpm, so I kept the check as-is.

Reply:
@lensbart we have a [`getPreferredPackageManager` utility](https://github.com/remix-run/remix/blob/dev/packages/remix-dev/cli/getPreferredPackageManager.ts) that you can use to check for the package manager.

Could you use that to enable the warnings for just `npm` and `yarn`?

Reply:
Consider it done! Except I’ve disabled the warnings for `yarn`, assuming that’s what you meant.